src/components/PeoplePage.tsx

```diff
@@ -1,12 +1,102 @@
+import { useEffect, useState } from 'react';
+import { useParams, useSearchParams } from 'react-router-dom';
 import { PeopleFilters } from './PeopleFilters';
 import { Loader } from './Loader';
 import { PeopleTable } from './PeopleTable';
+import { PersonType } from '../types';
+import { getPeople } from '../api';
+
+const peopleOnServer = (people: PersonType[]) => {
+  return people.map(person => {
+    const mother = people.find(pers => pers.name === person.motherName);
+    const father = people.find(pers => pers.name === person.fatherName);
+
+    return ({
+      ...person,
+      mother,
+      father,
+    });
+  });
+};
 
 export const PeoplePage = () => {
+  const [people, setPeople] = useState<PersonType[]>([]);
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState(false);
+  const [searchParams] = useSearchParams();
+  const { slug } = useParams();
+
+  useEffect(() => {
+    setLoading(true);
+    getPeople()
+      .then(peopleOnServer)
+      .then(preparedPeople => setPeople(preparedPeople))
+      .catch(() => setError(true))
+      .finally(() => setLoading(false));
+  }, []);
+
+  const preparedPeople = () => {
+    const sort = searchParams.get('sort');
+    const order = searchParams.get('order');
+    const sex = searchParams.get('sex');
+    const query = searchParams.get('query')?.trim().toLocaleLowerCase();
+    const centuries = searchParams.getAll('centuries');
+    let prepared = [...people];
+
+    if (query) {
+      prepared = prepared.filter(elem => elem.name.toLowerCase()
+        .includes(query)
+        || elem.motherName?.toLowerCase().includes(query)
+        || elem.fatherName?.toLowerCase().includes(query));
+    }
+
+    if (sex) {
+      prepared = prepared.filter(elem => elem.sex === sex);
+    }
+
+    if (centuries && centuries.length > 0) {
+      prepared = prepared.filter(elem => (
+        centuries?.includes(Math.ceil(elem.born / 100).toString())));
+    }
+
+    if (sort) {
+      switch (sort) {
+        case 'name':
+        {
+          prepared.sort((a, b) => a.name.localeCompare(b.name));
+          break;
+        }
+
+        case 'sex':
+        {
+          prepared.sort((a, b) => a.sex.localeCompare(b.sex));
+          break;
+        }
```

Comment:
```suggestion
        case 'name':
        case 'sex':
        {
          prepared.sort((a, b) => a[sort].localeCompare(b[sort]));
          break;
        }
```
you can group some cases